### PR TITLE
Pr v1alpha2 further refactorings

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
More refactorings for v1alpha2

This includes: (mostly syncs with CAPA)
* Logging (removed klog)/ event cleanup
* Makefile
* Switched to golangci linting (& fixed a lot of lint findings)
* Synced config folder
* Renamed keyname to sshKeyName
* Moved tools.go to hack/tools/
* Added some flags to the controller (like cluster & machine concurrency)
* Implement MachineDeployments

I updated the list of open issues for v1alpha2 here: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/380#issuecomment-526921003

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -142,71 +87,232 @@ test: generate lint ## Run tests
...
+	create cluster -v 4 \
+	--bootstrap-flags="name=clusterapi" \
+	--bootstrap-type kind \
+	-m ./examples/_out/controlplane.yaml \
+	-c ./examples/_out/cluster.yaml \
+	-p ./examples/_out/provider-components.yaml \
+	-a ./examples/addons.yaml
+
+.PHONY: create-cluster-management
+create-cluster-management: $(CLUSTERCTL) ## Create a development Kubernetes cluster on AWS in a KIND management cluster.
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply Makefile]: To much copy&paste from CAPA :). Fixed

File: config/default/manager_image_patch.yaml [hunk 2]
```diff
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: docker.io/sbueringer/cluster-api-provider-openstack:dev
+      - image: k8scloudprovider/cluster-api-openstack-controller-amd64:dev
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply config/default/manager_image_patch.yaml]: Yup let's use it for now. It's also the one which is published here: https://hub.docker.com/r/k8scloudprovider/openstack-cluster-api-controller/tags

(no idea how it's published to docker hub)

[the remark this answers is not in the document]
Comment (hidekazuna) [reply config/default/manager_image_patch.yaml]: The job in OpenLab publishes the images periodically.
https://github.com/theopenlab/openlab-zuul-jobs/blob/master/zuul.d/projects.yaml#L205

I agree to adjust when we discuss release process.

**Pull request merged**